modules/videoio/src/cap_v4l.cpp

```diff
@@ -824,34 +873,57 @@ bool CvCaptureCAM_V4L::createBuffers()
     size_t maxLength = 0;
     for (unsigned int n_buffers = 0; n_buffers < req.count; ++n_buffers) {
         v4l2_buffer buf = v4l2_buffer();
-        buf.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
+        v4l2_plane mplanes[VIDEO_MAX_PLANES];
+        size_t length;
+        off_t offset;
+        buf.type = type;
         buf.memory = V4L2_MEMORY_MMAP;
         buf.index = n_buffers;
+        if (V4L2_TYPE_IS_MULTIPLANAR(type)) {
+            buf.m.planes = mplanes;
+            buf.length = VIDEO_MAX_PLANES;
+        }
 
         if (!tryIoctl(VIDIOC_QUERYBUF, &buf)) {
             CV_LOG_WARNING(NULL, "VIDEOIO(V4L2:" << deviceName << "): failed VIDIOC_QUERYBUF: errno=" << errno << " (" << strerror(errno) << ")");
             return false;
         }
 
-        buffers[n_buffers].length = buf.length;
-        buffers[n_buffers].start =
-            mmap(NULL /* start anywhere */,
-                buf.length,
-                PROT_READ /* required */,
-                MAP_SHARED /* recommended */,
-                deviceHandle, buf.m.offset);
+        CV_Assert(1 < num_planes);
```

Comment:
The assert breaks behavior with existing cameras. Laplace example log for my Genius and Logitech webcam:
```
./example_cpp_laplace -c=0

This program demonstrates Laplace point/edge detection using OpenCV function Laplacian()
It captures from the camera of your choice: 0, 1, ... default 0
Call:
./example_cpp_laplace -c=<camera #, default 0> -p=<index of the frame to be decoded/captured next>

[ERROR:0@0.080] global /home/alexander/Projects/OpenCV/opencv-master/modules/videoio/src/cap.cpp (280) open VIDEOIO(V4L2): raised OpenCV exception:

OpenCV(4.6.0-dev) /home/alexander/Projects/OpenCV/opencv-master/modules/videoio/src/cap_v4l.cpp:892: error: (-215:Assertion failed) 1 < num_planes in function 'createBuffers'


Can't open camera/video stream: 0
```
The assertion should be with `<=`

Reply:
Thank you for your review.

I fixed it and confirmed the operation with both i.MX 8M (multi-planar) and UVC camera (single-planar).

Reply:
It’s my fault. I used the latest for the base, so there are a lot of diffs...

Check only the changes in cap_v4l.cpp.